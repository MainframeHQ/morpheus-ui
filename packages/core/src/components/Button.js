//@flow

import React, { Component } from 'react'
import styled, { css } from 'styled-components/native'

import { turnIntoField, type FieldProps } from '@morpheus-ui/forms'
import Theme from '../theme'

type Props = FieldProps & {
  title?: string,
  Icon?: any,
  HoverIcon?: any,
  iconSize: { width: number, height: number },
  backgroundColor?: string,
  backgroundHoverColor?: string,
  iconPosition?: 'top' | 'right' | 'left' | 'bottom',
  titleColor?: string,
  titleHoverColor?: string,
  iconColor?: string,
  iconHoverColor?: string,
  borderColor?: string,
  borderHoverColor?: string,
  shadow?: boolean,
  hoverShadow?: boolean,
  noBorder?: boolean,
  submit?: boolean,

  onPress?: () => void,
}

type State = {
  isHover: boolean,
}

const Container = styled.View`
  align-items: center;
  flex-direction: row;
`

const Clicker = styled.TouchableWithoutFeedback``

const InnerContainer = styled.View`
  transition: all 0.3s;
  align-items: center;
  justify-content: center;
  flex-direction: row;
  padding: ${props => props.muitheme.spacing.basic}px;
  border-radius: ${props => props.muitheme.borders.roundness}px;

  ${props =>
    props.backgroundColor &&
    css`
      background-color: ${props.backgroundColor};
    `};

  ${props =>
    !props.noBorder &&
    css`
      border-color: ${props =>
        props.borderColor || props.muitheme.colors.main.border};
      border-width: ${props => props.muitheme.borders.width}px;
    `};

  ${props =>
    (props.iconPosition === 'top' || props.iconPosition === 'bottom') &&
    css`
      flex-direction: column;
    `};

  ${props =>
    props.shadow &&
    css`
        shadow-color: #000;
        shadow-offset: {width: 0, height: 0};
        shadow-opacity: 0.1;
        shadow-radius: 8;
        `};

  ${props => props.containerStyles};

  ${props =>
    props.isHover &&
    css`
      ${props =>
        !props.noBorder &&
        css`
          border-color: ${props =>
            props.borderHoverColor ||
            props.borderColor ||
            props.muitheme.colors.main.border};
        `};

      ${props =>
        props.hoverShadow &&
        css`
          shadow-color: #000;
          shadow-offset: {width: 0, height: 0};
          shadow-opacity: 0.1;
          shadow-radius: 8;
          `};

      ${props =>
        props.backgroundHoverColor &&
        css`
          background-color: ${props.backgroundHoverColor ||
            props.backgroundColor ||
            'transparent'};
        `};

      ${props => props.containerHoverStyles};
    `};
`

const IconContainer = styled.View`
  transition: all 0.3s
  align-items: center;
  justify-content: center;
  ${props =>
    props.iconColor &&
    css`
      color: ${props.iconColor};
    `};

  ${props =>
    !!props.title &&
    props.iconPosition === 'top' &&
    css`
      margin-bottom: 10px;
    `};

  ${props =>
    !!props.title &&
    props.iconPosition === 'right' &&
    css`
      margin-left: 20px;
    `};

  ${props =>
    !!props.title &&
    props.iconPosition === 'bottom' &&
    css`
      margin-top: 10px;
    `};

  ${props =>
    !!props.title &&
    props.iconPosition === 'left' &&
    css`
      margin-right: 20px;
    `};

  ${props =>
    props.isHover &&
    css`
      ${props =>
        (props.iconHoverColor || props.iconColor) &&
        css`
          color: ${props.iconHoverColor || props.iconColor};
        `};
      ${props => props.iconHoverStyles};
    `};
`

const Title = styled.Text`
  font-family: 'Muli';
  font-size: ${props => props.muitheme.typography.fontSize - 2}px;
  color: #a9a9a9;
  transition: all 0.3s;
  font-weight: bold;
  color: ${props => props.titleColor || props.muitheme.colors.main.text};
  ${props => props.titleStyles};

  ${props =>
    props.isHover &&
    css`
      color: ${props =>
        props.titleHoverColor ||
        props.titleColor ||
        props.muitheme.colors.main.activeText};
      ${props => props.titleHoverStyles};
    `};
`
export class Button extends Component<Props, State> {
  static contextType = Theme

  static defaultProps = {
    iconPosition: 'right',
    iconSize: {
      width: 24,
      height: 24,
    },
    titleColor: '#979797',
    titleHoverColor: '#DA1157',
    iconColor: '#979797',
    iconHoverColor: '#DA1157',
    borderColor: '#979797',
    borderHoverColor: '#DA1157',
  }

  state = {
    isHover: false,
  }

  onMouseOver = () => {
    this.setState({
      isHover: true,
    })
  }

  onMouseLeave = () => {
    this.setState({
      isHover: false,
    })
  }

  onSubmit = () => {
    if (this.props.inForm && this.props.submit) {
      this.props.submitForm()
    }
    this.props.onPress && this.props.onPress()
  }

  renderIcon() {
    const { Icon, HoverIcon } = this.props
    if (!Icon) return null

    const TheIcon = this.state.isHover ? HoverIcon || Icon : Icon
    return (
      <IconContainer {...this.props} isHover={this.state.isHover}>
        <TheIcon
          width={this.props.iconSize.width}
          height={this.props.iconSize.height}
        />
      </IconContainer>
    )
  }

  render() {
    const { title, iconPosition } = this.props

    return (
      <Container>
        <Clicker onPress={this.onSubmit}>
          <InnerContainer
            {...this.props}
            onMouseOver={this.onMouseOver}
            onMouseLeave={this.onMouseLeave}
            muitheme={this.context}
            isHover={this.state.isHover}>
            {(iconPosition === 'left' || iconPosition === 'top') &&
              this.renderIcon()}
            <Title
              {...this.props}
              isHover={this.state.isHover}
              muitheme={this.context}>
              {title}
            </Title>
            {(iconPosition === 'right' || iconPosition === 'bottom') &&
              this.renderIcon()}
          </InnerContainer>
        </Clicker>
      </Container>
    )
  }
}

export default turnIntoField(Button)
