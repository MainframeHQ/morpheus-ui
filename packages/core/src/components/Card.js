// @flow

import React, { Component } from 'react'
import { View, StyleSheet } from 'react-native-web'

export default class Card extends Component<{}> {
  render() {
    return <View {...this.props} style={[styles.container, styles.flexi]} />
  }
}

const styles = StyleSheet.create({
  container: {
    width: 50,
    height: 50,
    backgroundColor: 'red',
  },
  flexi: {
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
  },
})
