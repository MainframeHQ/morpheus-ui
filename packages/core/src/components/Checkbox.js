// @flow

import React, { Component } from 'react'
import styled from 'styled-components/native'

import memoize from 'memoize-one'

import { turnIntoField, type FieldProps } from '@morpheus-ui/forms'

import transition from '../transitionClass'
import Theme, { getTheme } from './ThemeProvider'

const SvgCheckSymbol = props => (
  <svg width="11" height="7" viewBox="0 0 11 7" {...props}>
    <path
      d="M9.422.146L4 5.568.853 2.422a.5.5 0 0 0-.707.707l3.5 3.5a.498.498 0 0 0 .707 0L10.13.853a.5.5 0 0 0-.707-.707z"
      fill="currentColor"
      fillRule="evenodd"
    />
  </svg>
)

const OutterContainer = styled.View`
  flex-direction: row;
  align-items: center;
`

const Container = styled.View`
  flex-direction: column;
`
const Touchable = styled.TouchableWithoutFeedback``

const InnerContainer = styled.View`
  margin: ${props => props.muitheme.margin};
  flex-direction: row;
  align-items: center;
`

const CheckboxContainer = styled.View`
  background-color: ${props => props.muitheme.backgroundColor};
  padding: 1px;
  border-width: ${props => props.muitheme.containerBorderWidth};
  border-color: ${props => props.muitheme.containerBorderColor};
  border-style: solid;
  border-radius: ${props => props.muitheme.containerBorderRadius};
  align-items: center;
  justify-content: center;

  
  ${props =>
    !props.value &&
    props.muitheme.containerShadow &&
    `
    shadow-color: #000;
    shadow-offset: {width: 0, height: 0};
    shadow-opacity: 0.1;
    shadow-radius: 8;
  `}

    ${props =>
      props.value &&
      props.muitheme.containerCheckedShadow &&
      `
    shadow-color: #000;
    shadow-offset: {width: 0, height: 0};
    shadow-opacity: 0.1;
    shadow-radius: 8;
  `}

  ${props =>
    props.value &&
    `background-color: ${props.muitheme.backgroundCheckedColor};
     border-width: ${props.muitheme.containerCheckedBorderWidth};
     border-color: ${props.muitheme.containerCheckedBorderColor};
    `}

  ${props =>
    props.disabled &&
    `background-color: ${props.muitheme.backgroundDisabledColor};`}
`
const CheckSymbol = styled.Text`
  display: flex;
  align-items: center;
  justify-content: center;
  text-align: center;
  opacity: 0;
  width: ${props => props.muitheme.checkSymbolSize};
  height: ${props => props.muitheme.checkSymbolSize};
  color: ${props => props.muitheme.checkSymbolColor};


    ${props =>
      props.value &&
      props.muitheme.checkSymbolCheckedShadow &&
      `
    shadow-color: #000;
    shadow-offset: {width: 0, height: 0};
    shadow-opacity: 0.1;
    shadow-radius: 8;
  `}

  ${props => props.value && `opacity: 1;`}

  ${props =>
    props.disabled && `color: ${props.muitheme.checkSymbolDisabledColor};`}
`

const Label = styled.Text`
  font-family: ${props => props.muitheme.fontFamily};
  margin-left: ${props => props.muitheme.labelMargin};
  font-size: ${props => props.muitheme.fontSize};
  font-weight: ${props => props.muitheme.fontWeight};
  font-style: ${props => props.muitheme.fontStyle};
  color: ${props => props.muitheme.labelColor};

  ${props => props.value && `color: ${props.muitheme.labelCheckedColor};`}

  ${props => props.disabled && `color: ${props.muitheme.labelDisabledColor};`}
`

const ErrorMessage = styled.Text`
  font-family: ${props => props.muitheme.fontFamily};
  font-size: ${props => props.muitheme.errorFontSize};
  color: ${props => props.muitheme.errorColor};
  padding: 2px 0;
  margin-bottom: 2px;
`

type Props = FieldProps & {
  checkedLabel?: string,
  value?: ?boolean,
  defaultValue?: ?boolean,
  disabled: boolean,
  variant?: string | Array<string>,
}

type State = {
  value: boolean,
}

export class CheckboxComponent extends Component<Props, State> {
  static contextType = Theme

  constructor(props: Props) {
    super(props)

    this.state = {
      value: !!props.defaultValue || false,
    }
  }

  getCheckboxTheme = memoize((props: Props, context: Object) =>
    getTheme('Checkbox', props, context),
  )

  onPress = () => {
    if (!this.props.disabled) {
      const value = this.getValue()

      this.props.onChange && this.props.onChange(!value)
      !this.props.dirty && this.props.setDirty && this.props.setDirty()

      if (!this.props.inForm) {
        this.setState({
          value: !value,
        })
      }
    }
  }

  getValue() {
    const { inForm, defaultValue, value, fieldValue } = this.props

    if (inForm) {
      return fieldValue != null ? fieldValue : defaultValue || false
    }

    if (value != null) {
      return value || defaultValue || false
    }

    return this.state.value
  }

  render() {
    const muitheme: Object = this.getCheckboxTheme(this.props, this.context)
    const value = this.getValue()
    const showError =
      (this.props.isSubmitted() || this.props.dirty) &&
      !!this.props.errorMessage

    return (
      <OutterContainer>
        <Container>
          <Touchable disabled={this.props.disabled} onPress={this.onPress}>
            <InnerContainer muitheme={muitheme}>
              <CheckboxContainer
                disabled={this.props.disabled}
                muitheme={muitheme}
                className={transition}
                value={value}>
                <CheckSymbol
                  disabled={this.props.disabled}
                  muitheme={muitheme}
                  className={transition}
                  value={value}>
                  <SvgCheckSymbol />
                </CheckSymbol>
              </CheckboxContainer>
              {(this.props.checkedLabel || this.props.label) && (
                <Label
                  disabled={this.props.disabled}
                  muitheme={muitheme}
                  className={transition}
                  value={value}>
                  {value
                    ? this.props.checkedLabel || this.props.label
                    : this.props.label}
                </Label>
              )}
            </InnerContainer>
          </Touchable>
          <ErrorMessage muitheme={muitheme}>
            {showError ? this.props.errorMessage : ''}
          </ErrorMessage>
        </Container>
      </OutterContainer>
    )
  }
}

export default turnIntoField(CheckboxComponent)
