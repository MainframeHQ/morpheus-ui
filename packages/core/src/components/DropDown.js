// @flow

import React, { Component, createRef, type ElementRef } from 'react'
import styled from 'styled-components/native'

import memoize from 'memoize-one'

import { turnIntoField, type FieldProps } from '@morpheus-ui/forms'
import { Modal } from 'react-overlays'

import zIndex from '../zIndexClass'
import getElementPosition from '../getElementPosition'
import Theme, { getTheme } from './ThemeProvider'

const BACKSTYLE = {
  position: 'fixed',
  top: 0,
  left: 0,
  bottom: 0,
  right: 0,
  overflow: 'auto',
}

const SvgArrowDown = props => (
  <svg width="8" height="5" viewBox="0 0 8 5" {...props}>
    <path
      d="M7.146.146L4 3.293.853.146a.5.5 0 0 0-.707.708l3.5 3.5a.498.498 0 0 0 .708 0l3.5-3.5a.5.5 0 0 0-.708-.708z"
      fill="currentColor"
      fillRule="evenodd"
    />
  </svg>
)

const OutterContainer = styled.View`
  position: relative;
  margin: ${props => props.muitheme.margin};
`

const Container = styled.View`
  flex-direction: column;
  position: relative;
`
const Touchable = styled.TouchableOpacity`
  ${props => props.isopen && ` outline: none;`}
`

const InnerContainer = styled.View`
  position: relative;
  z-index: 1;
  min-width: ${props => props.muitheme.minWidth};
  align-items: center;
  justify-content: space-between;
  flex-direction: row;
  align-items: center;
  border-width: ${props => props.muitheme.borderWidth};
  border-color: ${props => props.muitheme.borderColor};
  border-style: solid;
  border-radius: ${props => props.muitheme.borderRadius};
  padding: ${props => props.muitheme.padding};
  background-color: ${props => props.muitheme.backgroundColor};

  ${props =>
    props.disabled &&
    `border-color: ${props.muitheme.borderDisabledColor};
     background-color: ${props.muitheme.backgroundDisabledColor};`}
`

const Label = styled.Text`
  font-family: ${props => props.muitheme.fontFamily};
  font-size: ${props => props.muitheme.fontSize};
  font-weight: ${props => props.muitheme.fontWeight};
  font-style: ${props => props.muitheme.fontStyle};
  color: ${props => props.muitheme.labelColor};

  ${props => props.value && `color: ${props.labelCheckedColor};`}

  ${props =>
    props.disabled &&
    `color: ${props.muitheme.labelDisabledColor};`}
 
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
`

const IconContainer = styled.Text`
  color: ${props => props.muitheme.arrowColor};
  display: flex;
  margin-left: ${props => props.muitheme.arrowMargin};
`

const ErrorMessage = styled.Text`
  font-family: ${props => props.muitheme.fontFamily};
  font-size: ${props => props.muitheme.errorFontSize};
  color: ${props => props.muitheme.errorColor};
  padding: 2px 0;
  margin-bottom: 2px;
`
const DropMenu = styled.View`
  background-color: ${props => props.muitheme.menuBackgroundColor};
  position: absolute;
  width: ${props => props.width}px;
  left: ${props => props.position.x}px;
  top: ${props => props.position.y - 1}px;
  ${props =>
    props.muitheme.menuShadow &&
    `
    shadow-color: #000;
    shadow-offset: {width: 0, height: 0};
    shadow-opacity: 0.1;
    shadow-radius: 8;
  `};

  max-height: ${props => props.muitheme.menuMaxHeight};

  border-width: ${props => props.muitheme.menuBorderWidth};
  border-color: ${props => props.muitheme.menuBorderColor};
  border-style: solid;
  border-radius: ${props => props.muitheme.menuBorderRadius};
`

const Backdrop = styled.TouchableOpacity`
  position: absolute;
  top: 0;
  right: 0;
  left: 0;
  bottom: 0;
  background-color: ${props => props.muitheme.menuBackdropColor};
`

const ScrollView = styled.ScrollView``

const DropItem = styled.TouchableOpacity`
  outline: none;
`

const ItemText = styled.Text`
  background-color: ${props => props.muitheme.menuItemBackgroundColor};
  padding: ${props => props.muitheme.menuItemPadding};
  color: ${props => props.muitheme.menuItemTextColor};
  font-size: ${props => props.muitheme.menuItemFontSize};
  border-bottom-width: ${props => props.muitheme.menuItemSeparatorWidth};
  border-bottom-color: ${props => props.muitheme.menuItemSeparatorColor};
  border-bottom-style: solid;

  ${props =>
    props.selected &&
    `background-color: ${props.muitheme.menuItemSelectedBackgroundColor};`}

  ${props =>
    props.withfocus === 'true' &&
    `background-color: ${props.muitheme.menuItemFocusBackgroundColor};
     color: ${props.muitheme.menuItemFocusTextColor};`}

  ${props =>
    props.selected && `color: ${props.muitheme.menuItemSelectedTextColor};`}
`

type Props = FieldProps & {
  checkedLabel?: string,
  value?: ?string,
  defaultValue?: string,
  disabled: boolean,
  variant?: string | Array<string>,
  options: Array<string | Object>,
  valueKey?: string,
  displayKey?: string,
}

type State = {
  open: boolean,
  value: ?string,
  focus: number,
  containerPos?: Object,
  containerWidth?: ?Number,
}

export class DropDownComponent extends Component<Props, State> {
  static contextType = Theme

  // $FlowFixMe: React Ref
  containerRef: ElementRef<'div'> = createRef()

  constructor(props: Props) {
    super(props)

    this.state = {
      value: (props.defaultValue && props.defaultValue.toString()) || null,
      open: false,
      focus: -1,
    }
  }

  componentDidMount() {
    this.getPosition()

    window.addEventListener('scroll', this.getPosition)
    window.addEventListener('resize', this.getPosition)
  }

  componentWillUnmount() {
    window.removeEventListener('scroll', this.getPosition)
    window.removeEventListener('resize', this.getPosition)
  }

  getPosition = () => {
    this.setState({
      containerPos: getElementPosition(this.containerRef.current),
      containerWidth: this.containerRef.current
        ? // $FlowFixMe: OffSet
          this.containerRef.current.offsetWidth
        : null,
    })
  }

  getDropDownTheme = memoize((props: Props, context: Object) =>
    getTheme('DropDown', props, context),
  )

  onPress = () => {
    if (this.state.open) {
      this.closeMenu()
    } else {
      this.openMenu()
    }
  }

  openMenu = () => {
    if (!this.props.disabled) {
      this.setState({ open: true })
      !this.props.dirty && this.props.setDirty && this.props.setDirty()
    }
  }

  closeMenu = () => {
    if (this.state.focus > -1) {
      const item = this.props.options[this.state.focus]
      if (typeof item === 'string') {
        this.selectItem(item)
      } else {
        this.selectItem(item[this.props.valueKey])
      }
    }
    this.dismissMenu()
  }

  dismissMenu = () => {
    this.setState({ open: false, focus: -1 })
  }

  handleInputKeyDown = (e: Object) => {
    const key = e.which || e.keyCode
    if (this.state.open) {
      if (key === 27) {
        // ESCAPE
        this.dismissMenu()
      }
      if (key === 38) {
        // Arrow UP
        this.moveFocus(-1)
      }
      if (key === 40) {
        // Arrow Down
        this.moveFocus(1)
      }
    } else {
      if (key === 38 || key === 40) {
        // Arrow UP or Down
        this.openMenu()
      }
    }
  }

  printValue = (value: any) => {
    const item = this.getItemFromValue(value)
    return this.printItem(item)
  }

  printItem = (item: string | Object) => {
    if (typeof item === 'string') {
      return item
    }

    if (!this.props.displayKey) {
      return item[this.props.valueKey]
    }

    return item[this.props.displayKey]
  }

  getItemFromValue = (value: string) => {
    if (!this.props.valueKey) {
      return value
    }

    return this.props.options.filter(item =>
      typeof item === 'string'
        ? item === value
        : // eslint-disable-next-line eqeqeq
          item[this.props.valueKey] == value,
    )[0]
  }

  getValueFromItem = (item: string | Object) => {
    if (typeof item === 'string') {
      return item
    }

    if (!this.props.valueKey) {
      return item
    }

    return item[this.props.valueKey]
  }

  getValue() {
    const { inForm, defaultValue, value, fieldValue } = this.props

    if (inForm) {
      return fieldValue != null ? fieldValue : defaultValue || null
    }

    if (value != null) {
      return value || defaultValue || null
    }

    return this.state.value
  }

  selectItem = (value: string) => {
    this.props.onChange && this.props.onChange(value)
    if (!this.props.inForm) {
      this.setState({
        value: value,
      })
    }
  }

  setFocus = (index: number) => {
    this.setState({ focus: index })
  }

  moveFocus = (offset: number) => {
    let nextFocusIndex = this.state.focus + offset

    if (nextFocusIndex < 0) {
      nextFocusIndex = this.props.options.length - 1
    } else if (nextFocusIndex >= this.props.options.length) {
      nextFocusIndex = 0
    }
    this.setFocus(nextFocusIndex)
  }

  render() {
    const muitheme: Object = this.getDropDownTheme(this.props, this.context)
    const value = this.getValue()
    const showError =
      (this.props.isSubmitted() || this.props.dirty) &&
      !!this.props.errorMessage

    return (
      <OutterContainer muitheme={muitheme} onKeyDown={this.handleInputKeyDown}>
        <div ref={this.containerRef}>
          <Container>
            <Touchable
              isopen={this.state.open}
              disabled={this.props.disabled}
              onPress={this.onPress}>
              <InnerContainer
                muitheme={muitheme}
                disabled={this.props.disabled}>
                <Label disabled={this.props.disabled} muitheme={muitheme}>
                  {value != null ? this.printValue(value) : this.props.label}
                </Label>
                {!this.props.disabled && (
                  <IconContainer muitheme={muitheme}>
                    <SvgArrowDown />
                  </IconContainer>
                )}
              </InnerContainer>
            </Touchable>
            <ErrorMessage muitheme={muitheme}>
              {showError ? this.props.errorMessage : ''}
            </ErrorMessage>
          </Container>
        </div>
        {this.state.open && (
          <>
            <Modal className={zIndex} show style={BACKSTYLE}>
              <div>
                <Backdrop
                  muitheme={muitheme}
                  onPress={this.dismissMenu}
                  onMouseOver={() => this.setFocus(-1)}
                />
                <DropMenu
                  muitheme={muitheme}
                  position={this.state.containerPos}
                  width={this.state.containerWidth}>
                  <ScrollView>
                    {this.props.options.map((item, index) => {
                      const itemValue = this.getValueFromItem(item)
                      return (
                        <DropItem
                          muitheme={muitheme}
                          onMouseOver={() => this.setFocus(index)}
                          onFocus={() => this.setFocus(index)}
                          onPress={() => this.closeMenu()}
                          key={`item-${index}`}>
                          <ItemText
                            muitheme={muitheme}
                            withfocus={(this.state.focus === index).toString()}
                            // $FlowFixMe
                            selected={value == itemValue}>
                            {this.printItem(item)}
                          </ItemText>
                        </DropItem>
                      )
                    })}
                  </ScrollView>
                </DropMenu>
              </div>
            </Modal>
          </>
        )}
      </OutterContainer>
    )
  }
}

export default turnIntoField(DropDownComponent)
