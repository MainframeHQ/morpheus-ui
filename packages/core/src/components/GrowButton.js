// @flow
import React, { Component } from 'react'
import styled, {
  css,
  type ReactComponentStyled,
} from 'styled-components/native'
import { View } from 'react-native'
import plusSymbol from '../../src/components/images/plus-symbol.svg'
import crossSymbol from '../../src/components/images/cross.svg'

type ContainerProps = {
  icon?: String,
  isHover: boolean,
}

const Container: ReactComponentStyled<ContainerProps> = styled.View`
  cursor: pointer;
  border-radius: 23px;
  padding: 10px 13px;
  border: 1px white solid;
  background-color: transparent;
  background-size: 12px 12px;
  background-position: 7px center;
  background-repeat: no-repeat;
  ${props =>
    props.icon === 'plus' &&
    css`
      background-image: url(${plusSymbol});
    `};
  ${props =>
    props.icon === 'cross' &&
    css`
      background-image: url(${crossSymbol});
    `};
  ${props =>
    props.isHover &&
    css`
      padding: 3px 15px 4px 25px;
    `};
`

const HoverText = styled.Text`
  display: none;
  ${props =>
    props.isHover &&
    css`
      display: block;
      color: white;
      font-size: 12px;
    `};
`

type Props = {
  hover?: String,
  icon?: String,
  onPress?: Function,
}

type State = {
  isHover: boolean,
}

export default class GrowButton extends Component<Props, State> {
  state = {
    isHover: false,
  }

  onMouseOver = () => {
    this.setState({
      isHover: true,
    })
  }

  onMouseLeave = () => {
    this.setState({
      isHover: false,
    })
  }

  render() {
    const { hover, onPress, icon } = this.props
    return (
      <View onMouseOver={this.onMouseOver} onMouseLeave={this.onMouseLeave}>
        <Container
          onClick={onPress}
          isHover={this.state.isHover}
          icon={icon ? icon : 'plus'}>
          <HoverText isHover={this.state.isHover}>
            {hover ? hover : 'Install'}
          </HoverText>
        </Container>
      </View>
    )
  }
}
