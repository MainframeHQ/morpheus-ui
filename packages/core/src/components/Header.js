// @flow

import React, { Component } from 'react'
import { View } from 'react-native-web'
import styled from 'styled-components'

type Props = {
  header: string,
  subhead?: string,
}

const HeaderStyle = styled.text`
  font-size: 5em;
  text-align: left;
  color: #1f3464;
  font-family: Helvetica, sans-serif;
  font-weight: 500;
  margin-bottom: 20px;
`

const SubHeaderStyle = styled.text`
  font-size: 1.5em;
  text-align: left;
  color: #1f3464;
  font-family: Helvetica, sans-serif;
  font-weight: 500;
  margin-top: 0;
`

export default class Header extends Component<Props> {
  render() {
    if (this.props.subhead) {
      return (
        <View>
          <HeaderStyle>{this.props.header}</HeaderStyle>
          <SubHeaderStyle>{this.props.subhead}</SubHeaderStyle>
        </View>
      )
    } else {
      return (
        <View>
          <HeaderStyle>{this.props.header}</HeaderStyle>
        </View>
      )
    }
  }
}
