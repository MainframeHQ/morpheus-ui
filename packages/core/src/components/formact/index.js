// @flow

export { default as Form } from './Form'
export { default as turnIntoField } from './Field'
