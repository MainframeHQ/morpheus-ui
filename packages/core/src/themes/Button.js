// Button Styles
export default {
  default: {
    iconPosition: 'right',
    iconWidth: 24,
    iconHeight: 24,
    iconMargin: 0,
    iconColor: '#979797',
    iconBackgroundColor: '#FFF',
    iconDisabledBackgroundColor: '#FFF',
    iconHoverBackgroundColor: '#DA1157',
    iconHoverColor: '#FFF',
    iconDisabledColor: '#D3D3D3',

    titleColor: '#979797',
    titleHoverColor: '#DA1157',
    titleDisabledColor: '#D3D3D3',

    borderRadius: 5,
    borderWidth: 1,
    borderColor: '#979797',
    borderHoverColor: '#DA1157',
    borderDisabledColor: '#D3D3D3',

    backgroundColor: '#FFF',
    backgroundHoverColor: '#FFF',
    backgroundDisabledColor: '#FFF',

    shadow: false,
    hoverShadow: false,
    iconPadding: 10,
    titlePadding: '10px',
    fontSize: 13,
    fontFamily: 'Muli',
    fontWeight: 'bold',
  },
  large: {
    fontSize: 15,
    iconWidth: 30,
    iconHeight: 30,
  },
  small: {
    fontSize: 10,
    iconWidth: 18,
    iconHeight: 18,
  },
  'icon-top': {
    iconPosition: 'top',
    iconMargin: 10,
  },
  'no-border': {
    borderWidth: 0,
    iconPadding: 0,
    titlePadding: '10px',
  },
  'hover-shadow': {
    hoverShadow: true,
    iconPadding: 10,
    titlePadding: '10px',
  },
}
