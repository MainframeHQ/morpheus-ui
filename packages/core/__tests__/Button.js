import React from 'react'

import { Button } from '../src/components/Button'

describe('Button', () => {
  it('renders', () => {
    expect(<Button title="Hello test" />).toMatchSnapshot()
  })
})
