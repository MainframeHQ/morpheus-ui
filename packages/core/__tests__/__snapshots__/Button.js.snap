// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders 1`] = `
<Button
  borderColor="#979797"
  borderHoverColor="#DA1157"
  iconColor="#979797"
  iconHoverColor="#DA1157"
  iconPosition="right"
  iconSize={
    Object {
      "height": 24,
      "width": 24,
    }
  }
  title="Hello test"
  titleColor="#979797"
  titleHoverColor="#DA1157"
/>
`;
